libcontainer/process_linux.go

```diff
@@ -420,7 +420,24 @@ func (p *initProcess) terminate() error {
 	if _, werr := p.wait(); err == nil {
 		err = werr
 	}
-	return err
+	return ignoreTerminateErrors(err)
+}
+
+// ignoreTerminateErrors returns nil if the given err matches an error known
+// to indicate that the terminate occurred successfully or err was nil, otherwise
+// err is returned unaltered.
+func ignoreTerminateErrors(err error) error {
+	if err == nil {
+		return nil
+	}
+
+	// TODO(steve): Update these to none string checks if the runtime exports them.
+	switch err.Error() {
+	case "os: process already finished", "exec: Wait was already called":
```

Comment:
We've requested the process be terminated, reporting an error due to the fact that the process had already gone is something the caller simply isn't interested in, as the desired effect has been achieved.

Its analogous to a method to remove a file reporting "file doesn't exist", the caller simply doesn't care.

To bring in some context:
1. When the init process fails to exec ```execSetns``` fails.
2. When ```execSetns``` fails it calls ```p.cmd.Wait()``` so the process has gone before return.
3. ```newParentProcess``` returns the error.
4. The cleanup code in container ```start()``` ensures everything is cleaned up by calling ```terminate```.
5. ```terminate``` returns the bogus error ```os: process already finished``` which is then logged.